src/main/java/io/aiven/kafka/connect/opensearch/RecordConverter.java

```diff
@@ -102,16 +111,20 @@ public DocWriteRequest<?> convert(final SinkRecord record, final String index) {
             }
         }
 
-        final DocumentIDStrategy docIdStrategy = config.docIdStrategy(record.topic());
-        
+        final var docIdStrategy = config.docIdStrategy(record.topic());
+
         if (Objects.isNull(record.value())) {
             return docIdStrategy.updateRequest(new DeleteRequest(index), record);
         }
 
-        final String payload = getPayload(record);
+        final var payload = getPayload(record);
```

Comment:
Please extract `getIndexRequest(record)` method so the decisions related to `config.dataStreamEnabled()` could be done in one single place.

Reply:
I think it needs to be done a bit diff. Since I'm implementing UPSERT write methon for indexes. I will address you comment in the next PR. WDYT?

Reply:
Why do we need next PR? I would suggested to have a clean implementation, this is no pressure anyway